File: my-workout-api/src/exercise-muscle/exercise-muscle.entity.ts
import { Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
import { ExercisesEntity } from '../exercises/exercises.entity';
import { MuscleEntity } from '../muscle/muscle.entity';

@Entity({ name: 'exercise_muscle', schema: 'public' })
export class ExerciseMuscleEntity {
    @PrimaryGeneratedColumn('increment')
    id: number;

    @ManyToOne(() => ExercisesEntity, (exercise) => exercise.exerciseMuscle)
    @JoinColumn({ name: 'exercise_id' })
    exercise: ExercisesEntity;

    @ManyToOne(() => MuscleEntity, (muscle) => muscle.exerciseMuscle)
    @JoinColumn({ name: 'muscle_id' })
    muscle: MuscleEntity;
}

File: my-workout-api/src/main/java/com/avernet/myworkoutapi/exercisemuscle/ExerciseMuscleEntity.java
package com.avernet.myworkoutapi.exercisemuscle;

import com.avernet.myworkoutapi.exercise.ExerciseEntity;
import com.avernet.myworkoutapi.muscle.MuscleEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "exercise_muscle")
@Data
public class ExerciseMuscleEntity {
    @Id
    Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "exercise_id")
    ExerciseEntity exercise;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "muscle_id")
    MuscleEntity muscle;
}
